### DISCUSSION TITLE
Nginx proxy manager + Heimdall access

### DISCUSSION THREAD

**Discussion by Ataraxiall** [linuxserver/Heimdall#735] [Ideas]
good morning

I am starting to use heimdall + Nginx proxu manager, each in their docker.

Ngnix has the option of creating users with their passwords to request them when entering the web.
When configured like this, it gives you the option to send the authorization to the host (heimdall in this case).
Is it possible that using the ngnix login it will pass it to Heimdall if it matches an already created username and password?

Anyway, what I ask is a bit strange ...
Thank you!

**Comment by ZHonry** [comment]
I have the same problem.

**Reply by emarcelcom** [reply]
I have the same problem, basically nginx proxy manager is giving me  the error: 502 Bad Gateway openresty, can't access Heimdall dashboard on top T-Pot Ce... traefik is working fine...

**Comment by aventi-lasith** [comment]
Anyone solved this issue?

**Comment by bigwen12** [comment]
The Same issue here....

**Comment by audas** [comment]
From what I can see this wont work behind a reverse_proxy which is pretty ridiculous. Will not work in caddy, will not work in a docker network even accessing directly via IP:PORT and the docs state you need to put in http(s)://DOCKERNAME:PORT to access docker containers.

It genuinely seems to be the most crazily badly designed piece of software.

**Comment by j0nnymoe** [comment]
OP question is nothing to do with reverse proxy but how it handles authentication. Other question was they were getting a 502 error which is a fairly common issue to solve with reverse proxies. 

As mentioned, Heimdall works fine with reverse proxy (works great with swag - https://github.com/linuxserver/reverse-proxy-confs/blob/master/heimdall.subdomain.conf.sample)

**Reply by audas** [reply]
Does not.

This example is not only 8 months old - but also fails to address even port 80 - the issue is the requirement for docker network id and port requirements which directly alludes to deeper issues in how it comms via networks.

Errors presented in issues backs this up.

Ngnx within the container is possibly the cause - but also how the internal php accesses things through guzzle etc being blocked by chrome..

Maybe just set up caddy and test - only takes a minute.


EDIT:

Also what are these.

        include /config/nginx/proxy.conf;
        include /config/nginx/resolver.conf;
  include /config/nginx/ssl.conf;

**Reply by j0nnymoe** [reply]
Those are links to configs we have within swag, all fairly generic stuff. 
Just to show, the container works fine (before any of the reverse proxy stuff): https://ci-tests.linuxserver.io/linuxserver/heimdall/latest/index.html 

You've not provided any details about how you've deployed the container/how you've setup your caddy instance.

**Reply by audas** [reply]
Ok.

Deleted and moved on.  Testing 50-60 dockers against  proxies. This one only one which fails.  

Can see where this is going. 

Edit:

And to be more clear - this is a basic page that shows links. Thats it. Homarr, Homepage, Flame and half a dozen more  offer vastly more complicated options without any issue at all.
